openfl/transport/grpc/director_server.py

```diff
@@ -197,15 +197,13 @@ async def GetExperimentData(self, request, context):  # NOQA:N802
                     break
                 yield director_pb2.ExperimentData(size=len(data), npbytes=data)
 
-    async def WaitExperiment(self, request_iterator, context):  # NOQA:N802
+    async def WaitExperiment(self, request, context):  # NOQA:N802
         """Request for wait an experiment."""
-        logger.info('Request WaitExperiment has got!')
-        async for msg in request_iterator:
-            logger.info(msg)
-            experiment_name = await self.director.wait_experiment(msg.collaborator_name)
-            logger.info(f'Experiment {experiment_name} was prepared')
+        logger.info(f'Request WaitExperiment has got from envoy {request.collaborator_name}!')
+        experiment_name = await self.director.wait_experiment(request.collaborator_name)
+        logger.info(f'Experiment {experiment_name} is ready for {request.collaborator_name}')
 
-            yield director_pb2.WaitExperimentResponse(experiment_name=experiment_name)
```

Comment:
What was the original thought behind this stream? If I remember correctly, our workspace archives were too big for a single message. Has something changed?

Reply:
Initially yes, but now an envoy received archives using GetExperimentData. And that API doesn't need in stream.